static/app/types/user.tsx

```diff
@@ -76,14 +76,18 @@ interface BaseApiToken {
   state: string;
 }
 
-// We include the token for API tokens used for internal apps
+// API Tokens should not be using and storing the token values in the application, as the tokens are secrets.
 export interface InternalAppApiToken extends BaseApiToken {
   application: null;
   refreshToken: string;
-  token: string;
   tokenLastCharacters: string;
 }
 
+// We include the token for new API tokens
+export interface NewInternalAppApiToken extends InternalAppApiToken {
```

Comment:
similar to the BE comment

Can we go the opposite way, and instead name this to `InternalAppApiToken` and replace the old one with `DeprecatedInternalAppApiToken`?

This way - rather than requiring another PR to both delete _and_ rename, we can just hash out a PR to delete the deprecated one?

Reply:
This won't be deleted! We need to capture and show the full API token value the first time the user makes the request, which means it should only be shown when the token is new, which is why I went with the `New` prefix haha. None of the work on this PR is temporary, and should be considered final fix! Does that help with why it's named like that, and the inheritance?

Reply:
Yup exactly hahah